radio_handler.h

```diff
@@ -0,0 +1,36 @@
+#ifndef RADIO_HANDLER_H_
+#define RADIO_HANDLER_H_
+
+#include "message_types.h"
+#include <stdint.h>
+
+/*
+ * Returns the state that the operators on the ground are telling us that the
+ * injector valve should be in. If we haven't heard from the operators ever (we
+ * just booted up and haven't established radio contact), this function will
+ * return VALVE_UNK, because we shouldn't assume that we should start with the
+ * injector valve in either state.
+ *
+ * If we haven't heard from the operator in TIME_NO_CONTACT_BEFORE_SAFE_STATE,
+ * this function will continue to spit out whatever value we last got from the
+ * operator, because going to safe state does not imply doing anything to the
+ * injector valve.
+ */
+enum VALVE_STATE radio_get_expected_inj_valve_state(void);
+
+/*
+ * Returns the state that the operators on the ground are telling us that the
+ * vent valve should be in. If we haven't heard from the operators ever (we
+ * just booted up and haven't established radio contact), this function will
+ * return VALVE_OPEN, because in the presence of no information, we should
+ * always open the vent valve.
+ *
+ * If we haven't heard from the operator in TIME_NO_CONTACT_BEFORE_SAFE_STATE,
+ * this function will return VALVE_OPEN, because that's the safest state for the
+ * vent valve to be in
```

Comment:
Please include a check for how long it's been since we received a full state command over the radio, and if it's been longer than TIME_NO_CONTACT_BEFORE_SAFE_STATE, this should return VALVE_OPEN.

There's a `millis()` function on master that returns milliseconds since bootup. If you merge master into this branch you can use that function

Reply:
Fixed in 8c79fd3.